### PR TITLE
Refactor range_rule

### PR DESCRIPTION
**Pull request by alandefreitas**

### CODE REVIEW

File: example/mailto/rfc.hpp [hunk 1]
```diff
@@ -330,13 +394,29 @@ constexpr auto addr_spec_rule =
...
-            addr_spec_rule));
+struct to_rule_t {
+    using value_type = urls::string_view;
+
+    urls::result< value_type >
+    parse(
+        char const*& it,
+        char const* end) const noexcept
+    {
+        static auto r = grammar::range_rule(
```
Comment (vinniefalco) [inline example/mailto/rfc.hpp:405]: This could be `const`

File: example/mailto/rfc.hpp [hunk 2]
```diff
@@ -330,13 +394,29 @@ constexpr auto addr_spec_rule =
...
 
 /// Rule for to = addr-spec *("," addr-spec )
-constexpr auto to_rule =
-    grammar::range_rule(
-        addr_spec_rule,
-        grammar::tuple_rule(
-            grammar::squelch(
-                grammar::delim_rule(',')),
-            addr_spec_rule));
+struct to_rule_t {
```
Comment (vinniefalco) [inline example/mailto/rfc.hpp:397]: is your IDE putting the brace on the same line?

File: example/mailto/rfc.hpp [hunk 3]
```diff
@@ -139,13 +139,29 @@ constexpr auto atext_token =
...
+    parse(
+        char const*& it,
+        char const* end) const noexcept
+    {
+        static auto r = grammar::range_rule(
+            atext_token,
+            grammar::tuple_rule(
+                grammar::squelch(
+                    grammar::delim_rule('.')),
+                atext_token));
```
Comment (vinniefalco) [inline example/mailto/rfc.hpp:155]: I would like to know what the compiler is emitting for the double-checked locking, if any

Comment (alandefreitas) [reply example/mailto/rfc.hpp:156]: Are there any good tools to do that locally? I'm only used to doing that with compiler explorer for small snippets but this PR is not going to be available there.

Comment (vinniefalco) [reply example/mailto/rfc.hpp:156]: I just set a breakpoint in visual studio and examine the assembly

**Comment by alandefreitas** [comment]
Stale

**Pull request closed without merging**